pkg/cloud/services/autoscaling/autoscalinggroup.go

```diff
@@ -526,6 +526,12 @@ func (s *Service) SubnetIDs(scope *scope.MachinePoolScope) ([]string, error) {
 		for _, subnet := range out.Subnets {
 			subnetIDs = append(subnetIDs, *subnet.SubnetId)
 		}
+
+		if len(subnetIDs) == 0 {
```

Comment:
Hm. So what happens if in the above there are filters and IDs defined? It might be that the list is not empty so the filter might not return anything, but the list still has a subnet ID in it.

I imagine we want that to proceed though and not care about the filter not working, right?

Reply:
Yeah in that scenario I'd expect it just to make use of the subnets with the matching IDs.